canvas_modules/common-canvas/src/toolbar/toolbar.scss

```diff
@@ -35,6 +37,40 @@ $toolbar-divider-width: 1px;
 
 	/* Prevent elements from being dragged onto canvas */
 	-webkit-user-drag: none;
+
+	&.toolbar-sm-div {
+		height: $toolbar-button-sm-height;
+		.toolbar-item {
+			& button {
+				padding: 0;
+				height: $toolbar-button-sm-height;
+				width: $toolbar-button-sm-width;
+				min-height: 30px;
+			}
+
+			&.default,
+			&.ghost {
+				& button {
+					height: $toolbar-button-sm-height;
+					width: $toolbar-button-sm-width;
+				}
+
+			}
+
+			&.tertiary {
+				& button {
+					height: $toolbar-button-sm-height;
+					width: $toolbar-button-sm-width;
+				}
+			}
+		}
+		.toolbar-spacer {
+			& button {
+				height: $toolbar-button-sm-height;
+				width: $toolbar-button-sm-width;
+			}
+		}
+	}
 }
```

Comment:
The Carbon buttons we use to display icons in the toolbar have a size prop which can be set through the button interface. Rather than overriding the sizes in the CSS (as above), can you try to see if you can pass in the toolbar's size value to the buttons that make up the toolbar and see if that has the desired effect? This would mean passing the size prop down to the lower level react objects that wrapper the  carbon buttons. What I am suggesting may not work because there may be other constraints on the size -- but we should try it and see if it works because that is better than overriding the button's CSS styles.  

One quick way to try this out (before you alter the code to pass the size parameter down through the react objects) is to hard code the size parameter for the carbon buttons we create in the palette code - then run the test harness and see what it looks like.

Reply:
Removed the overwrite CSS & adjusted the size prop.